coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -94,29 +97,53 @@ func TestAcquireJobWithCancel_Cancel(t *testing.T) {
 func TestHeartbeat(t *testing.T) {
 	t.Parallel()
 
-	numBeats := 3
-	ctx := testutil.Context(t, testutil.WaitShort)
-	heartbeatChan := make(chan struct{})
-	heartbeatFn := func(hbCtx context.Context) error {
-		t.Logf("heartbeat")
-		select {
-		case <-hbCtx.Done():
-			return hbCtx.Err()
-		default:
-			heartbeatChan <- struct{}{}
-			return nil
-		}
-	}
+	var (
+		ctx, cancel    = context.WithTimeout(context.Background(), testutil.WaitShort)
+		mClock         = clock.NewMock(t)
+		trapHbLoop     = mClock.Trap().NewTicker("heartbeat_loop")
+		trapHbLastSeen = mClock.Trap().Now("update_provisioner_daemon_last_seen_at")
+		ctrl           = gomock.NewController(t)
+		mStore         = dbmock.NewMockStore(ctrl)
```

Comment:
this test might be easier to understand with a `dbmem` Store, where you trigger a heartbeat, then call `GetProvisionerDaemons` and assert on `LastSeenAt`.

You'll need to convert the heartbeat ticker to `TickerFunc` so that when you `mClock.Advance(d).MustWait(ctx)`, you know the heartbeat is done processing.  With the regular ticker you only wait for the time to be sent on the channel, not that the other side of the channel is done processing.